### PR TITLE
Updates for operator sdk v1.28

### PR DESCRIPTION
**Pull request by michael-valdron**
**Please specify the area for this PR**

**What does does this PR do / why we need it**:

Provides updates to the project files to support Operator SDK version 1.28, these changes are documented under https://sdk.operatorframework.io/docs/upgrading-sdk-version/ up to [v1.28 docs](https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.28.0/).

Additional Changes:

- OpenShift Integration test script deploys 15 seconds before running test suite
- Integration test suite uses `registry-operator-system` for registry operator pod instead of `TEST_NAMESPACE`
  - This fixes the ability to use `TEST_NAMESPACE` as an optional separate namespace for the devfile registry test deployments

**Which issue(s) this PR fixes**:

Fixes #?

fixes devfile/api#881
fixes devfile/api#1106

**PR acceptance criteria**:

- [x] Test Coverage 
    - Are your changes sufficiently tested, and are any applicable test cases added or updated to cover your changes?
- [x] Gosec scans 
  - Fix all MEDIUM and higher findings and/or annotate a finding per gosec instructions: https://github.com/securego/gosec#annotating-code to address why a finding is not a security issue

Documentation
- [ ] Does the [registry operator documentation](https://github.com/devfile/devfile-web/tree/main/libs/docs/src/docs/no-version) need to updated with your changes?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: config/rbac/service_account.yaml [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
 metadata:
-  name: default
+  labels:
+    app.kubernetes.io/name: serviceaccount
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/component: rbac
+    app.kubernetes.io/created-by: devfileregistry-operator
+    app.kubernetes.io/part-of: devfileregistry-operator
+    app.kubernetes.io/managed-by: kustomize
+  name: sa
```
Comment (johnmcollier) [inline config/rbac/service_account.yaml:11]: Nit: `service-account` might be preferable to `sa` here.

**Review by johnmcollier — CHANGES_REQUESTED** [review]
Left some comments.

Also, could we move back the controller go sources from `internal/controller/xyz` to just `controllers/`? I think `internal/controller` is the default for new controller runtime projects, but the added "internal" in the path is unnecessary for our use case IMO

**Comment by michael-valdron** [comment]
> @michael-valdron: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> Test name 	Commit 	Details 	Required 	Rerun command
> ci/prow/v4.12-registry-operator-integration-test 	[1f0abf2](https://github.com/devfile/registry-operator/commit/1f0abf227101dd4cbada1fcd7559c46298c08477) 	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/devfile_registry-operator/45/pull-ci-devfile-registry-operator-main-v4.12-registry-operator-integration-test/1696183645716353024) 	true 	`/test v4.12-registry-operator-integration-test`
> 
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=devfile&repo=registry-operator&pr=45). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:michael-valdron).

CI is still running the integration tests using `go 1.18`, this needs to be updated to `go 1.19` for this to pass.

```
go: go.mod file indicates go 1.19, but maximum version supported by tidy is 1.18
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Change Log
```
[the remark this answers is not in the document]
Comment (michael-valdron) [reply CHANGELOG.md:1]: Yeah this was my thought too, at the moment its expecting manual entry but its something we should automate going forward. I can add this as an acceptance criteria for PRs until we do.

[the remark this answers is not in the document]
Comment (michael-valdron) [reply CHANGELOG.md:1]: Done: https://github.com/michael-valdron/devfile-registry-operator/blob/ff1108401205cdc55eb69b3506f80d26706301dd/.github/PULL_REQUEST_TEMPLATE.md?plain=1#L11

**Comment by michael-valdron** [comment]
> Left some comments.
> 
> Also, could we move back the controller go sources from `internal/controller/xyz` to just `controllers/`? I think `internal/controller` is the default for new controller runtime projects, but the added "internal" in the path is unnecessary for our use case IMO

Reverted go v4 project structure changes.

**Review by johnmcollier — APPROVED** [review]

**Review by johnmcollier — APPROVED** [review]
/lgtm

**Pull request merged**